cnd/src/db/save_message.rs

```diff
@@ -0,0 +1,353 @@
+};
+
+        diesel::insert_into(schema::rfc003_ethereum_bitcoin_erc20_bitcoin_request_messages::dsl::rfc003_ethereum_bitcoin_erc20_bitcoin_request_messages)
+            .values(&insertable)
+            .execute(&connection)
+            .map(|_| ())
+            .map_err(Error::Diesel)
+    }
+}
+
+#[derive(Insertable, Debug, Copy, Clone)]
+#[table_name = "rfc003_ethereum_bitcoin_accept_messages"]
+struct InsertableEthereumBitcoinAcceptMessage {
+    swap_id: Text<SwapId>,
+    ethereum_redeem_identity: Text<EthereumAddress>,
+    bitcoin_refund_identity: Text<bitcoin::PublicKey>,
+}
+
+impl_save_message! {
+    fn save_message(connection: SqliteConnection, message: Accept<Bitcoin, Ethereum>) -> Result<(), Error> {
+        let Accept {
+            id,
+            alpha_ledger_redeem_identity,
+            beta_ledger_refund_identity
+        } = message;
+
+        let insertable = InsertableEthereumBitcoinAcceptMessage {
+            swap_id: Text(id),
+            ethereum_redeem_identity: Text(EthereumAddress(beta_ledger_refund_identity)),
+            bitcoin_refund_identity: Text(alpha_ledger_redeem_identity.into_inner()),
+        };
+
+        diesel::insert_into(schema::rfc003_ethereum_bitcoin_accept_messages::dsl::rfc003_ethereum_bitcoin_accept_messages)
+            .values(&insertable)
+            .execute(&connection)
+            .map(|_| ())
+            .map_err(Error::Diesel)
+    }
+}
+
+#[derive(Insertable, Debug, Copy, Clone)]
+#[table_name = "rfc003_bitcoin_ethereum_accept_messages"]
+struct InsertableBitcoinEthereumAcceptMessage {
+    swap_id: Text<SwapId>,
+    bitcoin_redeem_identity: Text<bitcoin::PublicKey>,
+    ethereum_refund_identity: Text<EthereumAddress>,
+}
+
+impl_save_message! {
+    fn save_message(connection: SqliteConnection, message: Accept<Ethereum, Bitcoin>) -> Result<(), Error> {
+        let Accept {
+            id,
+            alpha_ledger_redeem_identity,
+            beta_ledger_refund_identity
+        } = message;
+
+        let insertable = InsertableBitcoinEthereumAcceptMessage {
+            swap_id: Text(id),
+            bitcoin_redeem_identity: Text(beta_ledger_refund_identity.into_inner()),
+            ethereum_refund_identity: Text(EthereumAddress(alpha_ledger_redeem_identity)),
+        };
+
+        diesel::insert_into(schema::rfc003_bitcoin_ethereum_accept_messages::dsl::rfc003_bitcoin_ethereum_accept_messages)
+            .values(&insertable)
+            .execute(&connection)
+            .map(|_| ())
+            .map_err(Error::Diesel)
+    }
+}
+
+#[derive(Insertable, Debug, Clone)]
+#[table_name = "rfc003_decline_messages"]
+struct InsertableDeclineMessage {
+    swap_id: Text<SwapId>,
+    reason: Option<String>,
+}
+
+impl_save_message! {
+    fn save_message(connection: SqliteConnection, message: Decline) -> Result<(), Error> {
+        let Decline {
+            id, reason: _reason // we don't map reason to a DB type because will be gone soon (hopefully)
```

Comment:
Can you open an issue so we can track it properly?

Reply:
Meh, really? :(

This stupid decline reason is in the way EVERYWHERE and we are going to remove it very soon hopefully. Not sure if it adds more noise to our backlog and we just waste time looking at it or if it has value to have it documented.